llama_index/vector_stores/elasticsearch.py

```diff
@@ -168,10 +172,11 @@ def __init__(
         self.batch_size = batch_size
         self.distance_strategy = distance_strategy
 
-        if es_client is not None:
+        if es_client is not None and async_es_client is not None:
             self._client = es_client
+            self._async_client = async_es_client
```

Comment:
I think it might be annoying/surprising for users to have to pass both the sync and async client. Curious: would it be possible to support only setting either sync or async, and we just support the corresponding methods, and raise errors on others?

Reply:
Alright, so yeah I agree it's annoying and surprising for user to have to pass both sync and async clients.

We can go for the approach you mentioned which is to support just one and raise errors on the others.

Another possibility is to use just `AsyncElasticsearch` and implement the sync methods `add`, `delete`, and `query` 
by calling sync versions of `async_add`, `adelete` and `aquery`, respectively. In this last push, I try this approach which is inpsired by Guido's take on this subject found [here](https://discuss.python.org/t/how-can-async-support-dispatch-between-sync-and-async-variants-of-the-same-code/15014/7).

All tests pass with this approach... tho even Guido says supporting both async/sync is a bit messy.... 

Let me know what you think 🤔

Reply:
Nice, love it! Let's go with this assuming there's no other gotchas. cc @sourabhdesai  for a quick look since he was doing some of the async support.